patroni/daemon.py

```diff
@@ -83,16 +83,18 @@ def abstract_main(cls, validator=None):
                         help='Patroni may also read the configuration from the {0} environment variable'
                         .format(Config.PATRONI_CONFIG_VARIABLE))
     args = parser.parse_args()
+    validate_config = validator and args.validate_config
     try:
-        if validator and args.validate_config:
+        if validate_config:
             Config(args.configfile, validator=validator)
             sys.exit()
 
         config = Config(args.configfile)
     except ConfigParseError as e:
         if e.value:
-            print(e.value)
-        parser.print_help()
+            print(e.value, file=sys.stderr)
```

Comment:
```suggestion
            sys.exit(e.value)
```

I think tests with python 2.7 are failing because of that.

Reply:
With the `sys.exit()` here, we'd no longer display help; is this okay?
Instead, I have used `from __future__ import print_function` to work around python2 hopefully.

Reply:
It also works.